libecl/src/ecl_file_view.c

```diff
@@ -0,0 +1,785 @@
+    fortio_fclose_stream( ecl_file_view->fortio );
+
+  return loadOK;
+}
+
+
+/*****************************************************************/
+
+
+
+void ecl_file_view_add_kw( ecl_file_view_type * ecl_file_view , ecl_file_kw_type * file_kw) {
+  if (ecl_file_view->owner)
+    vector_append_owned_ref( ecl_file_view->kw_list , file_kw , ecl_file_kw_free__ );
+  else
+    vector_append_ref( ecl_file_view->kw_list , file_kw);
+}
+
+void ecl_file_view_free( ecl_file_view_type * ecl_file_view ) {
+  vector_free( ecl_file_view->child_list );
+  hash_free( ecl_file_view->kw_index );
+  stringlist_free( ecl_file_view->distinct_kw );
+  vector_free( ecl_file_view->kw_list );
+  free( ecl_file_view );
+}
+
+void ecl_file_view_free__( void * arg ) {
+  ecl_file_view_type * ecl_file_view = ( ecl_file_view_type * ) arg;
+  ecl_file_view_free( ecl_file_view );
+}
+
+
+int ecl_file_view_get_num_named_kw(const ecl_file_view_type * ecl_file_view , const char * kw) {
+  if (hash_has_key(ecl_file_view->kw_index , kw)) {
+    const int_vector_type * index_vector = hash_get(ecl_file_view->kw_index , kw);
+    return int_vector_size( index_vector );
+  } else
+    return 0;
+}
+
+void ecl_file_view_fwrite( const ecl_file_view_type * ecl_file_view , fortio_type * target , int offset) {
+  int index;
+  for (index = offset; index < vector_get_size( ecl_file_view->kw_list ); index++) {
+    ecl_kw_type * ecl_kw = ecl_file_view_iget_kw( ecl_file_view , index );
+    ecl_kw_fwrite( ecl_kw , target );
+  }
+}
+
+
+
+
+int ecl_file_view_iget_occurence( const ecl_file_view_type * ecl_file_view , int global_index) {
+  const ecl_file_kw_type * file_kw = vector_iget_const( ecl_file_view->kw_list , global_index);
+  const char * header              = ecl_file_kw_get_header( file_kw );
+  const int_vector_type * index_vector = hash_get( ecl_file_view->kw_index , header );
+  const int * index_data = int_vector_get_const_ptr( index_vector );
+
+  int occurence = -1;
+  {
+    /* Manual reverse lookup. */
+    int i;
+    for (i=0; i < int_vector_size( index_vector ); i++)
+      if (index_data[i] == global_index)
+        occurence = i;
+  }
+  if (occurence < 0)
+    util_abort("%s: internal error ... \n" , __func__);
+
+  return occurence;
+}
+
+void ecl_file_view_fprintf_kw_list(const ecl_file_view_type * ecl_file_view , FILE * stream) {
+  int i;
+  for (i=0; i < vector_get_size( ecl_file_view->kw_list ); i++) {
+    const ecl_file_kw_type * file_kw = vector_iget_const( ecl_file_view->kw_list , i );
+    fprintf(stream , "%-8s %7d:%s\n",
+            ecl_file_kw_get_header( file_kw ) ,
+            ecl_file_kw_get_size( file_kw ) ,
+            ecl_util_get_type_name( ecl_file_kw_get_type( file_kw )));
+  }
+}
+
+
+ecl_file_view_type * ecl_file_view_alloc_blockview2(const ecl_file_view_type * ecl_file_view , const char * start_kw, const char * end_kw, int occurence) {
+  if ((start_kw == NULL) || ecl_file_view_get_num_named_kw( ecl_file_view , start_kw ) > occurence) {
```

Comment:
could you turn this into an early-return?

That is, if not that thing `(start_kw == NULL) || ecl_file_view_get_num_named_kw( ecl_file_view , start_kw ) > occurence`, return NULL?  Then everything else can be _unindented_